lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -204,12 +204,19 @@ static int trace_fragment_write(struct nrf_modem_trace_data *frag)
 
 		if (ret < 0) {
 			LOG_ERR("trace_backend.write failed with err: %d", ret);
-
 			return ret;
 		}
 
 		if (ret == 0) {
 			LOG_WRN("trace_backend wrote 0 bytes.");
+			/* On fault, modem reinitialization depends on the trace thread to complete
+			 * the processing of traces. In this case we want to prevent processing
+			 * traces forever. Therefore we abort if the trace backend write returns
+			 * zero when the modem is not initialized.
+			 */
+			if (!nrf_modem_is_initialized()) {
+				return -ESHUTDOWN;
+			}
 		}
```

Comment:
Could you tell in which situations the `write` function of the backend can return `0`? I think it should return either a positive value (on success) or a negative one (on error); `uart_send` in this file should never return `0`.

Reply:
Yes, ideally it should not return 0. I think I added this while developing the uart backend, when I saw that, if the backend were to return 0, it would block the modem shutdown/reinit. So, this is more a safeguard to be sure that this does not happen.